app/controllers/admin/merchants_controller.rb

```diff
@@ -8,20 +8,18 @@ def show
     @merchant = Merchant.find_by(id: params[:merchant_id])
   end
 
-  def disable
-    merchant = Merchant.find(params[:id])
-    merchant.items.each {|item| item.update_attribute(:active?, false)}
-    merchant.update_attribute(:active?, false)
-    redirect_to "/admin/merchants"
-    flash[:notice] = "disabled #{merchant.name}"
+  def update
+    @merchant = Merchant.find(params[:id])
+    if @merchant.active? == true
+      @merchant.items.each {|item| item.update_attribute(:active?, false)}
+      @merchant.update_attribute(:active?, false)
+      redirect_to "/admin/merchants"
+      flash[:notice] = "disabled #{@merchant.name}"
+    else
+      @merchant.items.each {|item| item.update_attribute(:active?, true)}
+      @merchant.update_attribute(:active?, true)
+      redirect_to "/admin/merchants"
+      flash[:notice] = "enabled #{@merchant.name}"
+    end
```

Comment:
![](https://media1.giphy.com/media/yzvVXSvrg7JxC/giphy.gif)

Reply:
Yay this is so wonderful! As discussed, looking forward to trying to implement ActiveRecord instead of the each block.

Reply:
chaching!